pkg/utils/utils_virtual.go

```diff
@@ -205,6 +226,33 @@ func getOpenstackDataFromMetadataService() (metaData *OSPMetaData, networkData *
 	return metaData, networkData, nil
 }
 
+// getPCIAddressFromMACAddress returns the PCI address of a device given its MAC address
+func getPCIAddressFromMACAddress(macAddress string) (string, error) {
+	nics, err := ghw.Network()
+	if err != nil {
+		return "", fmt.Errorf("error getting network info: %v", err)
+	}
+	// Check if there are more than one device with the same MAC address
+	// and return an error if that's the case, we don't support that scenario
+	// for now.
+	numberOfDevices := 0
```

Comment:
as noted on slack instead of looping over nics.NICs twice
you can instead declare a var here to hold the PCI address
then 

do
```
for _, nic := range nics.NICs {
		if nic.MacAddress == macAddress {
			if pciAddress == nil {
               pciAddress = nic.PCIAddress
            } else {
             return "", fmt.Errorf("more than one device found with MAC address %s", macAddress)
            }
		}
	}
if pciAdress != nil {
    return *nic.PCIAddress, nil
}
return "", fmt.Errorf("no device found with MAC address %s", macAddress)
```

Reply:
nice catch, thanks